src/scorer.coffee

```diff
@@ -178,13 +178,26 @@ scoreMain = (subject, subject_lw, prepQuery) ->
   mm = subject_lw.lastIndexOf(query_lw[n - 1], m)
   if(mm > i) then m = mm + 1
 
+  csc_invalid = true
+
   while ++i < m     #foreach char si of subject
+    si_lw = subject_lw[i]
+
+    # if si_lw is not in query
+    if not prepQuery.charCodes[si_lw.charCodeAt 0]?
```

Comment:
what about something like `if not si_lw.charCodeAt(0) of prepQuery.charCodes`

or in js `if !(si_lw.charCodeAt(0) in prepQuery.charCodes)`
vs checking `if(undefined==null)` ?